pkg/resources/secret_with_oauth_authorization_code_grant.go

```diff
@@ -0,0 +1,218 @@
+package resources
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/helpers"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/internal/logging"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/internal/provider"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/customdiff"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var secretAuthorizationCodeGrantSchema = func() map[string]*schema.Schema {
+	secretAuthorizationCodeGrant := map[string]*schema.Schema{
+		"oauth_refresh_token": {
+			Type:        schema.TypeString,
+			Required:    true,
+			Description: "Specifies the token as a string that is used to obtain a new access token from the OAuth authorization server when the access token expires.",
+		},
+		"oauth_refresh_token_expiry_time": {
+			Type:             schema.TypeString,
+			Required:         true,
+			DiffSuppressFunc: IgnoreChangeToCurrentSnowflakeValueInShow("oauth_refresh_token_expiry_time"),
+			Description:      "Specifies the timestamp as a string when the OAuth refresh token expires. Accepted string formats: YYYY-MM-DD, YYYY-MM-DD HH:MI, YYYY-MM-DD HH:MI:SS, YYYY-MM-DD HH:MI <timezone>",
+		},
+		"api_authentication": {
+			Type:             schema.TypeString,
+			ValidateDiagFunc: IsValidIdentifier[sdk.AccountObjectIdentifier](),
+			Required:         true,
```

Comment:
Add DiffSuppressFunc for ignoring quotes

Reply:
Added